simpeg/objective_function.py

```diff
@@ -435,57 +447,61 @@ def multipliers(self):
         list of int
             Multipliers for the objective functions.
         """
-        return self._multipliers
+        return [obj.multiplier for obj in self.objfcts]
 
     @multipliers.setter
-    def multipliers(self, value):
+    def multipliers(self, value: list[float] | None):
         """Set multipliers attribute after checking if they are valid."""
+        if value is None:
+            return
+
         for multiplier in value:
             _validate_multiplier(multiplier)
+
         _check_length_objective_funcs_multipliers(self.objfcts, value)
-        self._multipliers = value
+
+        for val, obj in zip(value, self.objfcts):
+            obj.multiplier = val
 
     def __call__(self, m, f=None):
         """Evaluate the objective functions for a given model."""
         fct = 0.0
-        for i, phi in enumerate(self):
-            multiplier, objfct = phi
-            if multiplier == 0.0:  # don't evaluate the fct
+        for i, objfct in enumerate(self):
+            if objfct.multiplier == 0.0:  # don't evaluate the fct
                 continue
             if f is not None and objfct.has_fields:
                 objective_func_value = objfct(m, f=f[i])
             else:
                 objective_func_value = objfct(m)
-            fct += multiplier * objective_func_value
+            fct += objfct.multiplier * objective_func_value
         return fct
 
     def deriv(self, m, f=None):
         # Docstring inherited from BaseObjectiveFunction
         g = Zero()
-        for i, phi in enumerate(self):
-            multiplier, objfct = phi
-            if multiplier == 0.0:  # don't evaluate the fct
+        for i, objfct in enumerate(self):
+            if objfct.multiplier == 0.0:  # don't evaluate the fct
                 continue
             if f is not None and objfct.has_fields:
                 aux = objfct.deriv(m, f=f[i])
             else:
                 aux = objfct.deriv(m)
             if not isinstance(aux, Zero):
-                g += multiplier * aux
+                g += objfct.multiplier * aux
```

Comment:
@dccowan @jcapriot @santisoler @lheagy 

The main change happens here, where the top combo asks for the function's multiplier, rather than storing a list of multipliers and zipping through both. So no change on the value of `function.call()` or `function.deriv()`. 

The main advantage is that directives can act on scaling functions directly, without having to worry about the order in the global combo. Beta simply becomes the `inv.reg.multiplier` also - all same mechanism for addition.

Reply:
@santisoler So we could call a specific method instead to get the same value scaled. If that makes more sense to you guys. Something like
```
g += objfct.scaled_deriv(m)
```

or as @thibaut-kobold suggested
```
g += objfct.deriv(m, scaled=True)
```
and we leave the default behaviour as scaled=False